charms/jupyter-ui/src/charm.py

```diff
@@ -170,9 +235,22 @@ def _configure_mesh(self, interfaces):
                 }
             )
 
+    def _is_container_ready(self):
+        """Check if connection can be made with container.
+
+        Returns: False if container is not available
+                 True if connection can be made
+        Sets maintenance status is container is not acvailable.
+        """
+        if not self.container.can_connect():
+            self.unit.status = MaintenanceStatus("Waiting for pod startup to complete")
+            return False
+        return True
```

Comment:
Would this feel more natural following the pattern of raising an ErrorWithStatus rather than directly setting `unit.status=MaintenanceStatus`?  I think it stands out as handling an error differently than the other parts of the code

If we changed this to raise an error, we'd also want to change [this line](https://github.com/canonical/notebook-operators/pull/97/files#diff-05545afb2aff573c9a874b983233f37c3bff993f5df47eb5317af9dc4c4af2d5R207)

Reply:
This method basically delegates decision to caller. Also, container not ready is not an error.